pkg/template/controller/templateinstance_controller.go

```diff
@@ -15,23 +19,29 @@ import (
 	"k8s.io/kubernetes/pkg/kubectl/resource"
 
 	"github.com/openshift/origin/pkg/api/latest"
-	authclient "github.com/openshift/origin/pkg/auth/client"
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
 	"github.com/openshift/origin/pkg/client"
 	"github.com/openshift/origin/pkg/config/cmd"
 	templateapi "github.com/openshift/origin/pkg/template/api"
 	templateapiv1 "github.com/openshift/origin/pkg/template/api/v1"
 	templateclientset "github.com/openshift/origin/pkg/template/generated/internalclientset"
 	internalversiontemplate "github.com/openshift/origin/pkg/template/generated/internalclientset/typed/template/internalversion"
+	kclientsetinternal "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
 )
 
 type TemplateInstanceController struct {
-	restconfig     rest.Config
+	oc             *client.Client
+	kc             kclientsetinternal.Interface
 	templateclient internalversiontemplate.TemplateInterface
 	controller     cache.Controller
 }
 
-func NewTemplateInstanceController(restconfig rest.Config) *TemplateInstanceController {
-	c := TemplateInstanceController{restconfig: restconfig, templateclient: templateclientset.NewForConfigOrDie(&restconfig).Template()}
+func NewTemplateInstanceController(restconfig *rest.Config) *TemplateInstanceController {
+	c := TemplateInstanceController{
+		oc:             client.NewOrDie(restconfig),
```

Comment:
To confirm: is the pattern that I should cause a templateapi client to be created somewhere higher up the call stack and pass this in with the oc & kc clients?